dapp/src/components/earn/CurveStake.js

```diff
@@ -0,0 +1,559 @@
+import React, { useState, useEffect } from 'react'
+import { fbt } from 'fbt-runtime'
+import { useWeb3React } from '@web3-react/core'
+import { useStoreState } from 'pullstate'
+import { ethers, Contract, BigNumber } from 'ethers'
+
+import withIsMobile from 'hoc/withIsMobile'
+import analytics from 'utils/analytics'
+import { formatCurrency } from 'utils/math'
+import ContractStore from 'stores/ContractStore'
+import CoinStore from 'stores/CoinStore'
+import addresses from 'constants/contractAddresses'
+
+// Just adding the methods we are using
+const gaugeMiniAbi = [
+  {
+    name: 'working_supply',
+    outputs: [{ type: 'uint256', name: '' }],
+    inputs: [],
+    stateMutability: 'view',
+    type: 'function',
+  },
+  {
+    name: 'inflation_rate',
+    outputs: [{ type: 'uint256', name: '' }],
+    inputs: [],
+    stateMutability: 'view',
+    type: 'function',
+  },
+  {
+    stateMutability: 'view',
+    type: 'function',
+    name: 'totalSupply',
+    inputs: [],
+    outputs: [{ name: '', type: 'uint256' }],
+  },
+  {
+    stateMutability: 'view',
+    type: 'function',
+    name: 'reward_data',
+    inputs: [{ name: 'arg0', type: 'address' }],
+    outputs: [
+      { name: 'token', type: 'address' },
+      { name: 'distributor', type: 'address' },
+      { name: 'period_finish', type: 'uint256' },
+      { name: 'rate', type: 'uint256' },
+      { name: 'last_update', type: 'uint256' },
+      { name: 'integral', type: 'uint256' },
+    ],
+  },
+]
+const gaugeControllerMiniAbi = [
+  {
+    name: 'gauge_relative_weight',
+    outputs: [{ type: 'uint256', name: '' }],
+    inputs: [{ type: 'address', name: 'addr' }],
+    stateMutability: 'view',
+    type: 'function',
+  },
+  {
+    name: 'gauge_relative_weight',
+    outputs: [{ type: 'uint256', name: '' }],
+    inputs: [
+      { type: 'address', name: 'addr' },
+      { type: 'uint256', name: 'time' },
+    ],
+    stateMutability: 'view',
+    type: 'function',
+  },
+]
+
+const CurveStake = ({ rpcProvider, isMobile }) => {
+  const { active } = useWeb3React()
+  const [baseApy, setBaseApy] = useState(false)
+  const [crvBaseApy, setCrvBaseApy] = useState(false)
+  const [crvBoostedApy, setCrvBoostedApy] = useState(false)
+  const [ognApy, setOgnApy] = useState(false)
+  const [totalBaseApy, setTotalBaseApy] = useState(false)
+  const [totalBoostedApy, setTotalBoostedApy] = useState(false)
+  const [curveRate, setCurveRate] = useState(false)
+  const [virtualPrice, setVirtualPrice] = useState(false)
+  const [gaugeContract, setGaugeContract] = useState(false)
+  const [gaugeControllerContract, setGaugeControllerContract] = useState(false)
+  const chainId = useStoreState(ContractStore, (s) => s.chainId)
+  const readOnlyProvider = useStoreState(
+    ContractStore,
+    (s) => s.readOnlyProvider
+  )
+  const ognPrice = useStoreState(CoinStore, (s) => s.ogn.price)
+
+  const fetchBaseApy = async () => {
```

Comment:
nit: This function can be a React Hook to keep the code cleaner

Reply:
Thanks good point, have separated that logic into a hook as suggested